docker-argocd-app-diff/argocd-recursive-sync.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+[ $# -lt 1 ] && echo "Usage $0 <root app> [--only-application]" && exit 1
+if [ "$2" = "--only-application" ]; then
+  echo "Recursive Sync (only application)..."
+  only_app=true
+else
+  echo "Recursive Sync..."
+  only_app=false
+fi
+
+apps_status=$(mktemp -d)
+
+update_sync_status () {
+  [ "$DEBUG" = 'true' ] && echo -n "Refresh app status..."
+  apps_manifests=$(mktemp -d)
+  kubectl get Application --all-namespaces -o yaml > $apps_manifests/apps.yaml
+
+  yq r $apps_manifests/apps.yaml 'items[*].metadata.name' > $apps_manifests/names.yaml
+  yq r $apps_manifests/apps.yaml 'items[*].status.sync.status' > $apps_manifests/status.yaml
```

Comment:
Shouldn't it be easier to output in json and use jq?

Reply:
yes, fixed with last commit: no more yq  usage and no more docker image